bellatrix.playwright/src/main/java/solutions/bellatrix/playwright/components/common/webelement/WebElement.java

```diff
@@ -0,0 +1,702 @@
+    }
+
+    public WebElement getByText(String text, GetByTextOptions options) {
+        if (isFrame) {
+            return new WebElement(wrappedFrameLocator.getByText(text, options.convertTo(FrameLocator.GetByTextOptions.class)));
+        }
+
+        return new WebElement(wrappedLocator.getByText(text, options.convertTo(Locator.GetByTextOptions.class)));
+    }
+
+    public WebElement getByText(String text) {
+        if (isFrame) {
+            return new WebElement(wrappedFrameLocator.getByText(text));
+        }
+
+        return new WebElement(wrappedLocator.getByText(text));
+    }
+
+    public WebElement getByText(Pattern text, GetByTextOptions options) {
+        if (isFrame) {
+            return new WebElement(wrappedFrameLocator.getByText(text, options.convertTo(FrameLocator.GetByTextOptions.class)));
+        }
+
+        return new WebElement(wrappedLocator.getByText(text, options.convertTo(Locator.GetByTextOptions.class)));
+    }
+
+    public WebElement getByText(Pattern text) {
+        if (isFrame) {
+            return new WebElement(wrappedFrameLocator.getByText(text));
+        }
+
+        return new WebElement(wrappedLocator.getByText(text));
+    }
+
+    public WebElement getByTitle(String text, GetByTitleOptions options) {
+        if (isFrame) {
+            return new WebElement(wrappedFrameLocator.getByTitle(text, options.convertTo(FrameLocator.GetByTitleOptions.class)));
+        }
+
+        return new WebElement(wrappedLocator.getByTitle(text, options.convertTo(Locator.GetByTitleOptions.class)));
+    }
+
+    public WebElement getByTitle(String text) {
+        if (isFrame) {
+            return new WebElement(wrappedFrameLocator.getByTitle(text));
+        }
+
+        return new WebElement(wrappedLocator.getByTitle(text));
+    }
+
+    public WebElement getByTitle(Pattern text, GetByTitleOptions options) {
+        if (isFrame) {
+            return new WebElement(wrappedFrameLocator.getByTitle(text, options.convertTo(FrameLocator.GetByTitleOptions.class)));
+        }
+
+        return new WebElement(wrappedLocator.getByTitle(text, options.convertTo(Locator.GetByTitleOptions.class)));
+    }
+
+    public WebElement getByTitle(Pattern text) {
+        if (isFrame) {
+            return new WebElement(wrappedFrameLocator.getByTitle(text));
+        }
+
+        return new WebElement(wrappedLocator.getByTitle(text));
+    }
+
+    /**
+     * Identical with the Locator.locator() method.
+     */
+    public WebElement locate(String selector) {
+        if (isFrame) {
+            return new WebElement(wrappedFrameLocator.locator(selector));
+        }
+
+        return new WebElement(wrappedLocator.locator(selector));
+    }
+
+    /**
+     * Identical with the Locator.locator() method.
+     */
+    public WebElement locate(WebElement webElement) {
+        if (isFrame) {
+            return new WebElement(wrappedFrameLocator.locator(webElement.wrappedLocator));
+        }
```

Comment:
I wonder why half of the methods have the isFrame logic, while the resot no not? 

Also, since it looks like having an element in a frame is a big deal in playwright, have you thought of creating a FrameWebElement that inherits from WebElement, and only oveerides the methods needed?

Reply:
FrameLocator in Playwright has methods only for finding elements inside the iframe. Other methods available for Locator, are not available for FrameLocator.
I made the WebElement treat differently iframes and other html elements when it comes to relatively finding other elements on the page, but treat the iframe as a normal html element when you want to use methods inherent to Locator, such as boundingBox(), click(), evaluate(), getAttribute(), scrollIntoViewIfNeeded().

Creating FrameWebElement that inherits from WebElement is a better idea than to have inline if-else logic. On the other hand, I want to make it available to the user to set the isFrame to true or false as per the user's needs.

When using FrameLocator.getBy methods, Playwright will search **inside** the iframe.
When using Locator.getBy methods, Playwright will search **near** the iframe.

Possible use case:
I want to find an iframe element, find something inside of it, and then find something near the iframe, on the same page, reusing the component/webelement.

By 'near' the iframe, I mean cases like:
```
<div>
 <iframe>
 <input>
</div>
```
and you want to do //iframe//following-sibling::input

Maybe better naming for the boolean isFrame, in such case, would be asFrame?

Reply:
I'll move the logic for checking if the component is Frame.class from WebComponent to the Frame.class itself.